src/lib/datatypes/sol-vector.c

```diff
@@ -349,18 +349,20 @@ sol_vector_del_element(struct sol_vector *v, const void *elem)
 SOL_API int
 sol_ptr_vector_del_element(struct sol_ptr_vector *pv, const void *elem)
 {
-    int r;
-    uint16_t i, removed = 0;
-    void *cur;
-
-    SOL_PTR_VECTOR_FOREACH_REVERSE_IDX (pv, cur, i)
-        if (cur == elem) {
-            r = sol_ptr_vector_del(pv, i);
-            SOL_INT_CHECK(r, < 0, r);
+    uint16_t i, removed = 0, offset = 0;
+    void **v = pv->base.data;
+
+    for (i = 0; i < pv->base.len; i++)
+        if (v[i] == elem)
             removed++;
-        }
+        else
+            v[offset++] = v[i];
```

Comment:
@barbieri, I'll do some profiling to confirm that. I'll also apply @lpereira suggestion on eliminating ```removed``` variable and then make the pull request v2.